src/main/groovy/nextflow/prov/renderers/GexfRenderer.groovy

```diff
@@ -0,0 +1,531 @@
+
+            w.writeEmptyElement("attvalue")
+            w.writeAttribute("for", ATT_FULLPATH)
+            w.writeAttribute("value", path.toString())
+
+            final fileSize = getFileSize()
+            if( fileSize >= 0L ) {
+                w.writeEmptyElement("attvalue")
+                w.writeAttribute("for", ATT_FILESIZE)
+                w.writeAttribute("value", "${fileSize}")
+            }
+
+            final fileType = getFileType()
+            if( fileType ) {
+                w.writeEmptyElement("attvalue")
+                w.writeAttribute("for", ATT_FILETYPE)
+                w.writeAttribute("value", fileType)
+            }
+
+            final fileTimestamp = getFileTimestamp()
+            if( fileTimestamp >= 0L ) {
+                w.writeEmptyElement("attvalue")
+                w.writeAttribute("for", ATT_FILETIMESTAMP)
+                w.writeAttribute("value", "${fileTimestamp}")
+            }
+
+            w.writeEndElement() // attvalues
+
+            w.writeEmptyElement("viz", "shape", VIZ_NS)
+            w.writeAttribute("value", "square")
+
+            w.writeEmptyElement("viz", "color", VIZ_NS)
+            w.writeAttribute("hex", "#CA3C66")
+
+            if( vizSize > 0 ) {
+                w.writeEmptyElement("viz", "size", VIZ_NS)
+                w.writeAttribute("value", "${vizSize}")
+            }
+
+            w.writeEndElement()
+        }
+    }
+
+
+    private static class GexfGraph {
+
+        private final Set<GexfNode> nodes = new HashSet<>()
+        private final Set<GexfEdge> edges = new HashSet<>()
+
+        GexfNode addNode(GexfNode n) {
+            nodes.add(n)
+            return n
+        }
+
+        GexfEdge addEdge(GexfNode source, GexfNode target) {
+            final e = new GexfEdge(source, target)
+            edges.add(e)
+            return e
+        }
+
+        void write(Path xmlFile) {
+            setVizSizes(nodes)
+
+            Files.newOutputStream(xmlFile).withCloseable { outputStream ->
+                write0(outputStream)
+            }
+        }
+
+        private void write0(OutputStream outputStream) {
+            final xof = XMLOutputFactory.newInstance()
+            final encoding = "UTF-8"
+            final w = xof.createXMLStreamWriter(outputStream, encoding)
+
+            w.writeStartDocument(encoding, "1.0")
+            w.writeStartElement("gexf")
+            w.writeAttribute("xmlns", XMLNS)
+            w.writeAttribute("xmlns:xsi", XMLConstants.W3C_XML_SCHEMA_INSTANCE_NS_URI)
+            w.writeAttribute("xmlns:viz", VIZ_NS)
+            w.writeAttribute("xsi:schemaLocation", SCHEMA_LOCATION)
+            w.writeAttribute("version", VERSION)
+            w.writeStartElement("meta")
+            w.writeAttribute("lastmodifieddate", new SimpleDateFormat("yyyy-MM-dd").format(new Date()))
+            w.writeStartElement("creator")
+            w.writeCharacters("GEXF Renderer for nf-prov by Pierre Lindenbaum")
```

Comment:
Are you sure you want your name in every GEXF file created by every nextflow run around the world? :sweat_smile: